src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +265,71 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+        try:
+            cdata = repository.get(id)
+        except Repository.ObjectNotFound:
+            print("object %s not found." % hex_id)
+            return EXIT_ERROR
+
```

Comment:
guess you duplicated this from `get-obj`?

maybe it is simpler to just use the data from the output file of `get-obj` than having this duplication.

consistency: you also do not put the object to the repo for `debug format-obj`

Reply:
Yes. `get-obj` writes to a file, hence I thought not to re-use the code (or otherwise I will have to write another function which gets the data but does not write the file and the modify the code.

Reply:
I didn't mean you call the code in get-obj from here.

What I meant is that the user first uses `borg debug get-obj` to write a borg repo object to a file.
Then that file can be transformed via `borg debug parse-obj` to the metadata and data file.